src/backends/cg.rs

```diff
@@ -45,24 +53,31 @@ impl<D: HasDisplayHandle, W: HasWindowHandle> SurfaceInterface<D, W> for CGImpl<
             RawWindowHandle::AppKit(handle) => handle,
             _ => return Err(InitError::Unsupported(window_src)),
         };
-        let view = handle.ns_view.as_ptr() as id;
-        let window: id = unsafe { msg_send![view, window] };
-        let window: id = unsafe { msg_send![window, retain] };
+
+        let mtm =
+            MainThreadMarker::new().expect("can only access AppKit handles on the main thread");
```

Comment:
Not sure if you want to handle this as a panic, as a `raw_window_handle::HandleError::Unavailable`, or something else?

Reply:
We should add a new type for the `Error` enum to the tune of "not on the main thread."

...actually, is `Error` `non_exhaustive`? It might be better to just return a `BackendError`.